src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -20,15 +22,21 @@ public class WebJobsScriptHostService : IHostedService, IScriptHostManager, IDis
         private readonly IServiceProvider _rootServiceProvider;
         private readonly IServiceScopeFactory _rootScopeFactory;
         private readonly IOptionsMonitor<ScriptApplicationHostOptions> _applicationHostOptions;
-        private readonly IScriptWebHostEnvironment _scriptWebHostEnvironment;
         private readonly ILogger _logger;
+        private readonly IEnvironment _environment;
+        private readonly HostPerformanceManager _performanceManager;
+        private readonly IOptions<HostHealthMonitorOptions> _healthMonitorOptions;
+        private readonly SlidingWindow<bool> _healthCheckWindow;
+        private readonly Timer _hostHealthCheckTimer;
+
         private IHost _host;
         private CancellationTokenSource _startupLoopTokenSource;
         private int _hostStartCount;
         private bool _disposed = false;
 
         public WebJobsScriptHostService(IOptionsMonitor<ScriptApplicationHostOptions> applicationHostOptions, IServiceProvider rootServiceProvider,
-            IServiceScopeFactory rootScopeFactory, IScriptWebHostEnvironment scriptWebHostEnvironment, ILoggerFactory loggerFactory)
+            IServiceScopeFactory rootScopeFactory, ILoggerFactory loggerFactory, IEnvironment environment,
+            HostPerformanceManager hostPerformanceManager, IOptions<HostHealthMonitorOptions> healthMonitorOptions)
```

Comment:
It seems like all of this health monitoring could move into its own IHostedService to keep things from intermingling so much in this service.